### PR TITLE
Aryan roots dev

### PR DESCRIPTION
**Pull request by Aryan-BanyanCloud**

### CODE REVIEW

File: components/input/textFieldv2/TextFieldv2.jsx [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+	};
+
+	const getType = (currType) => {
+		if (currType === 'email') {
+			setInputType('input');
+		}
+		setInputType(currType);
+	};
+
+	const getPlaceholder = (currPlaceholder) => {
```
Comment (PDP-Banyan) [inline components/input/textFieldv2/TextFieldv2.jsx:123]: This is not needed. We will not have predefined `placeholder` for the types. The placeholder is optional and will be passed from the application.

File: components/input/textFieldv2/TextFieldv2.jsx [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+		autocomplete,
+		autocompleteOptions,
+		custom,
+	} = props;
+
+	const { current: isControlled } = useRef(value !== undefined);
+
+	// for uncontrolled input
+	const [uncontrolledValue, setUncontrolledValue] = useState(defaultValue);
+	const [inputType, setInputType] = useState('textarea');
```
Comment (PDP-Banyan) [inline components/input/textFieldv2/TextFieldv2.jsx:92]: Here in the initialValue, you can rewrite this as below

`useState(type || 'textarea')`

or

```
useState(()=>{
   if(type){
       return type
   }
   return "textArea"
})
```

File: components/input/textFieldv2/TextFieldv2.jsx [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+		checkAndOpenAutocomplete(fieldValue);
+
+		if (isControlled) {
+			onChange(event, fieldValue);
+		} else {
+			setUncontrolledValue(fieldValue);
+		}
+	};
+
+	const getType = (currType) => {
```
Comment (PDP-Banyan) [inline components/input/textFieldv2/TextFieldv2.jsx:116]: If you implement the [above one](https://github.com/banyandata-cloud/roots/pull/624/files#r1699302025), you can remove this function and the useEffect from where it is called.

File: components/input/dropdownv2/Dropdownv2.jsx [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+								animate={{
+									opacity: 1,
+									scale: 1,
+								}}>
+								{multi && (
+									<li
+										ref={multiOptionsRef}
+										className={styles['multi-options']}
+										tabIndex={-1}>
+										{selectedOptions?.length > 0 ? (
```
Comment (PDP-Banyan) [inline components/input/dropdownv2/Dropdownv2.jsx:450]: Remove this condition.

File: components/input/dropdownv2/Dropdownv2.jsx [hunk 5]
```diff
@@ -0,0 +1,582 @@
...
+											size='auto'
+											onClick={onApply}
+										/>
+									</div>
+								)}
+							</motion.ul>
+						</FloatingFocusManager>
+					)}
+				</Popper>
+				{feedback != null && (
```
Comment (PDP-Banyan) [inline components/input/dropdownv2/Dropdownv2.jsx:524]: You can remove this, as we got rid of the old feedback below the input.

File: components/paginationv2/Paginationv2.module.scss [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+						height: 2.375rem;
+						max-height: 100%;
+						width: auto;
+						min-width: fit-content;
+						padding: 0.625rem;
+						border: 0rem solid transparent;
+						& > [data-elem='placeholder'] {
+							color: #1a2028;
+							font-size: 0.875rem;
+							background-color: #0f62fe;
```
Comment (PDP-Banyan) [inline components/paginationv2/Paginationv2.module.scss:65]: replace with variable name.

File: components/Toggle/Togglev2.module.scss [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		height: 100%;
+		background: transparent;
+
+		> button[data-elem='base-cell'].toggle-button {
+			position: relative;
+			z-index: 2;
+			background: transparent;
+			min-width: 5rem;
+			width: auto;
+			height: auto;
```
Comment (PDP-Banyan) [inline components/Toggle/Togglev2.module.scss:23]: Put the `border-radius` here, instead of duplicating the same in each sub level classes.

File: components/Toggle/Togglev2.module.scss [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+				&.highlight {
+					background-color: $mono-color1;
+				}
+				> [data-elem='component2'] {
+					> [data-elem='title'] {
+						color: $white;
+					}
+				}
+				&.secondary {
+					background-color: #cfdfff;
```
Comment (PDP-Banyan) [inline components/Toggle/Togglev2.module.scss:41]: Replace all hex-codes with variables

File: components/Toggle/Togglev2.jsx [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+		if (multi) {
+			return defaultValue ?? options?.[0]?.value ?? [];
+		}
+		return defaultValue ?? options?.[0]?.value;
+	});
+
+	const { current: isControlled } = useRef(value !== undefined);
+
+	// eslint-disable-next-line no-nested-ternary
+	const inputValue = isControlled ? value ?? '' : uncontrolledValue;
```
Comment (PDP-Banyan) [inline components/Toggle/Togglev2.jsx:47]: Convert this into code with let as below:

```
let inputValue = '';

if(isControlled){
    inputValue = value
} 
else{
      inputValue = uncontrolledValue;
}
```

File: components/Toggle/Togglev2.jsx [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+				{multi && (
+					<Button
+						type='button'
+						size='auto'
+						data-elem='toggle'
+						variant='text'
+						className={classes(
+							styles['toggle-button'],
+							// eslint-disable-next-line no-nested-ternary
+							allSelected
```
Comment (PDP-Banyan) [inline components/Toggle/Togglev2.jsx:136]: Try to convert this ternary condition in the same way as [this one](https://github.com/banyandata-cloud/roots/pull/624/files?file-filters%5B%5D=.jsx&file-filters%5B%5D=.scss&show-viewed-files=true#r1700621744)

File: components/Toggle/Togglev2.jsx [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+	fallback: PropTypes.bool,
+	className: PropTypes.string,
+	theme: PropTypes.oneOf(['dark', 'light']),
+	options: PropTypes.arrayOf(PropTypes.string),
+	value: PropTypes.string,
+	multi: PropTypes.bool,
+	smooth: PropTypes.bool,
+};
+
+Togglev2.defaultProps = {
```
Comment (PDP-Banyan) [inline components/Toggle/Togglev2.jsx:212]: Remove this and initiate the values in the props itself.

File: components/breadcrumbsv2/Breadcrumbs.jsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+import styles from './Breadcrumbs.module.css';
+import { CaretIcon, FlowChartIcon } from '../icons';
+
+const BreadCrumbs = (props) => {
+	const { crumbs, className } = props;
+
+	const [expand, setExpand] = useState(false);
+	const [anchorEl, setAnchorEl] = useState(null);
+
+	const CrumbsDOM =
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.jsx:17]: Write a condition to return `null`, if `crumbs` is null.

File: components/breadcrumbsv2/Breadcrumbs.jsx [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+				}`}
+				rightComponent={() => {
+					return (
+						<div className={classes(styles.flowchart)}>
+							<FlowChartIcon className={classes(expand ? styles.icon : '')} />
+						</div>
+					);
+				}}
+			/>
+			<Popover
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.jsx:81]: Don't open the popper when the crumbs length is equal to 1.

File: components/breadcrumbsv2/Breadcrumbs.jsx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+			title: PropTypes.string,
+			value: PropTypes.string,
+			path: PropTypes.string,
+			icon: PropTypes.node,
+		})
+	),
+	className: PropTypes.string,
+};
+
+BreadCrumbs.defaultProps = {
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.jsx:105]: Remove this defaultProps declaration and initiate the values in the props itself.

File: components/breadcrumbsv2/Breadcrumbs.module.scss [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+	button[data-elem='base-cell'].selected[type='submit'] {
+		@include flex(row, flex-start, center);
+		height: 1.9375rem;
+		width: auto;
+		background: #f7f7f8;
+		padding: 0.5rem 0.75rem;
+		gap: 0.625rem;
+		border-radius: 0.25rem;
+		&.expand {
+			border: 0.0313rem solid #0f62fe;
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.module.scss:17]: Replace all the hex-codes with variables

File: components/breadcrumbsv2/Breadcrumbs.module.scss [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+
+.root {
+	@include flex(row, flex-start, center);
+	width: 100%;
+	height: 100%;
+
+	button[data-elem='base-cell'].selected[type='submit'] {
+		@include flex(row, flex-start, center);
+		height: 1.9375rem;
+		width: auto;
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.module.scss:11]: Have max width as per design for both button and options

File: components/breadcrumbsv2/Breadcrumbs.module.scss [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+				height: 12px;
+				background-color: $secondary-color2;
+				border-radius: 50%;
+				margin-left: 8px;
+			}
+			.circle-dropdown::before {
+				content: '';
+				width: 0%;
+				height: 26px;
+				border-left: 2px dotted #007bff;
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.module.scss:120]: Use Secondary blue instead primary and replace hox-codes.

File: components/breadcrumbsv2/Breadcrumbs.jsx [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+							</div>
+						)
+					);
+				}}
+				title={`${crumbs?.[(crumbs?.length ?? 1) - 1]?.title} : ${
+					crumbs?.[(crumbs?.length ?? 1) - 1]?.value
+				}`}
+				rightComponent={() => {
+					return (
+						<div className={classes(styles.flowchart)}>
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.jsx:75]: You can remove this wrapper(`div`)

File: components/breadcrumbsv2/Breadcrumbs.module.scss [hunk 19]
```diff
@@ -0,0 +1,138 @@
+@import '../../styles/index';
+
+.root {
+	@include flex(row, flex-start, center);
+	width: 100%;
+	height: 100%;
+
+	button[data-elem='base-cell'].selected[type='submit'] {
+		@include flex(row, flex-start, center);
+		height: 1.9375rem;
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.module.scss:10]: Instead of using fixed height. Have that as min hight and use padding as per design

File: components/breadcrumbsv2/Breadcrumbs.module.scss [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+		}
+		> [data-elem='component1'] {
+			@include flex(row, flex-start, center);
+			div.iconWrapper {
+				@include flex(row, center);
+			}
+		}
+		> [data-elem='component2'] {
+			@include flex(row, center);
+			color: $text-color;
```
Comment (PDP-Banyan) [inline components/breadcrumbsv2/Breadcrumbs.module.scss:27]: Text should have ellipsis if big

**Comment by PDP-Banyan** [comment]
There were few missing styles and codes refactors done in all the input fields. I could see possible edge cases with respect to breadcrumb. Will see once implemented in the actual app. Meanwhile, you can test the input especially the textfield in the login page for behaviour @Aryan-BanyanCloud.

Will check the other PR after the integration tomorrow.  Proceeding the merge of this PR.

**Review by PDP-Banyan — APPROVED** [review]
